### ISSUE TITLE
CodeArtifact Role Assumption not producing correct workflow

### ISSUE THREAD

**Issue by jessebye** [projen/projen#1843]
When attempting to use a CodeArtifact NPM registry and assume a role, the generated release workflow still tries to use the Access key ID / Secret access key:

```
      - name: Release
        run: npx -p publib@latest publib-npm
        env:
          NPM_DIST_TAG: latest
          NPM_REGISTRY: my-domain.d.codeartifact.us-east-2.amazonaws.com/npm/npm/
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Here is the projen config being used:
```
import { cdk8s } from "projen";
const project = new cdk8s.ConstructLibraryCdk8s({
  ...
  codeArtifactOptions: {
    roleToAssume: "arn:aws:fake"
  },
  npmRegistryUrl: "https://my-domain.d.codeartifact.us-east-2.amazonaws.com/npm/npm/",
```

I tried to look in the code to see why this might be happening, but couldn't immediately identify any problem.

**Comment by Chriscbr** [comment]
Sounds like a possible bug.

@AminFazlMondo @charliehart-mondo can you take a look?

**Comment by AminFazlMondo** [comment]
I think this is the expected behaviour as the role assumption command, needs to have `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` to assume the role, and then will override them and also set the session token.
https://github.com/cdklabs/publib/blob/ef1aa5a52f4c74829e2a49d8fd104ba665b55027/bin/publib-npm#L30

Also, we are using similar projen project but our release code is similar to below. are you missing the `AWS_ROLE_TO_ASSUME` @jessebye 

```
      - name: Release
        run: npx -p publib@latest publib-npm
        env:
          NPM_DIST_TAG: latest
          NPM_REGISTRY: my-domain.d.codeartifact.us-east-2.amazonaws.com/npm/npm/
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          AWS_ROLE_TO_ASSUME: arn:aws:fake
```

**Comment by jessebye** [comment]
@AminFazlMondo yes, when we specify `roleToAssume`, projen gives us a Release step that is missing `AWS_ROLE_TO_ASSUME`. Could you share what your projen config looks like? Maybe I missed something in mine?

But there is another problem...

Even if we get `AWS_ROLE_TO_ASSUME` to populate there, we would need another pre-release step that uses [this action](https://github.com/aws-actions/configure-aws-credentials) to get the access key id/secret access key from the GitHub OIDC provider. I can put this step into `releaseWorkflowSetupSteps` but it ends up in the wrong job ("release" - but we need it in "release_npm"), which doesn't work because jobs don't share environment variables.

**Comment by AminFazlMondo** [comment]
Thanks @jessebye, had look at this a bit further and found out that there are two issues here:

1. We used `typescript.TypeScriptProject` in our repo and that generated `AWS_ROLE_TO_ASSUME`, and did not add that for `cdk8s.ConstructLibraryCdk8s`
2. As you mentioned this will not support GitHub OIDC with the currect implementation. The code is limited to provide the GitHub secret variable names. This could be extended , and I agree with you one of those approaches could be;  to allow for environament variables to be used in this step, and add a `Configure AWS credentials` step prior to release step

**Comment by AminFazlMondo** [comment]
Potential fix for the first issue: 
https://github.com/projen/projen/pull/1846

**Comment by jessebye** [comment]
@Chriscbr would you be able to reopen this issue? #1846 is only a partial fix for this issue.

**Comment by AminFazlMondo** [comment]
Hi @mrgrain and @jessebye ,
I think this has been fixed as part of #2399

**Comment by mrgrain** [comment]
Thanks @AminFazlMondo !

Everyone else, if you still find something not working please let me know here or create a new issue.
